onadata/apps/api/viewsets/tableau_viewset.py

```diff
@@ -0,0 +1,264 @@
+                        flat_dict.update(unpack_data_per_qstn_type(
+                            key, value, qstn_type=qstn_type))
+            result.append(dict(flat_dict))
+    return result
+
+
+class TableauViewSet(OpenDataViewSet):
+    flattened_dict = defaultdict(list)
+
+    @action(methods=['GET'], detail=True)
+    def data(self, request, **kwargs):
+        self.object = self.get_object()
+        # get greater than value and cast it to an int
+        gt_id = request.query_params.get('gt_id')
+        gt_id = gt_id and parse_int(gt_id)
+        count = request.query_params.get('count')
+        pagination_keys = [
+            self.paginator.page_query_param,
+            self.paginator.page_size_query_param
+        ]
+        query_param_keys = request.query_params
+        should_paginate = any([k in query_param_keys for k in pagination_keys])
+
+        data = []
+        if isinstance(self.object.content_object, XForm):
+            if not self.object.active:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            xform = self.object.content_object
+            if xform.is_merged_dataset:
+                qs_kwargs = {'xform_id__in': list(
+                    xform.mergedxform.xforms.values_list('pk', flat=True))}
+            else:
+                qs_kwargs = {'xform_id': xform.pk}
+            if gt_id:
+                qs_kwargs.update({'id__gt': gt_id})
+
+            # Filter out deleted submissions
+            instances = Instance.objects.filter(
+                **qs_kwargs, deleted_at__isnull=True).order_by('pk')
+
+            if count:
+                return Response({'count': instances.count()})
+
+            if should_paginate:
+                instances = self.paginate_queryset(instances)
+
+            data = process_tableau_data(
+                TableauDataSerializer(instances, many=True).data, xform)
+
+            return self.get_streaming_response(data)
+
+        return Response(data)
+
+    def flatten_xform_columns(
+            self, json_of_columns_fields, table: str = None):
+        '''
+        Flattens a json of column fields while splitting columns into separate
+        table names for each repeat and then sets the result to a class
+        variable
+        '''
+        for a in json_of_columns_fields:
+            table_name = table or 'data'
+            if a.get('type') != 'repeat':
+                self.flattened_dict[table_name].append(
+                    {
+                        'name': a.get('name'),
+                        'type': self.get_tableau_type(a.get('type'))})
+            else:
+                table_name = a.get('name')
+
+            # using IGNORED_FIELD_TYPES so that choice values are not included.
+            if a.get('children') and a.get('type') not in IGNORED_FIELD_TYPES:
+                self.flatten_xform_columns(a.get('children'), table=table_name)
+
+    def clean_xform_headers(self, headers: list) -> list:
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/onaio/onadata/pullRequest?prid=6260785)